fedimint-dummy-client/src/lib.rs

```diff
@@ -0,0 +1,386 @@
+    ) -> ClientOutput<<Self::Common as ModuleCommon>::Output, Self::States> {
+        // Construct output and state machine to track the tx
+        ClientOutput {
+            output: DummyOutput {
+                amount,
+                account: self.key.x_only_public_key().0,
+            },
+            state_machines: Arc::new(move |txid, _| {
+                vec![DummyStateMachine::Output(amount, txid, id)]
+            }),
+        }
+    }
+
+    async fn await_primary_module_output(
+        &self,
+        operation_id: OperationId,
+        _out_point: OutPoint,
+    ) -> anyhow::Result<Amount> {
+        let stream = self
+            .notifier
+            .subscribe(operation_id)
+            .await
+            .filter_map(|state| async move {
+                match state {
+                    DummyStateMachine::OutputDone(amount, _) => Some(Ok(amount)),
+                    DummyStateMachine::Refund(_) => Some(Err(anyhow::anyhow!(
+                        "Error occurred processing the dummy transaction"
+                    ))),
+                    _ => None,
+                }
+            });
+
+        pin_mut!(stream);
+
+        stream.next_or_pending().await
+    }
+
+    async fn get_balance(&self, dbtc: &mut ModuleDatabaseTransaction<'_>) -> Amount {
+        get_funds(dbtc).await
+    }
+
+    async fn subscribe_balance_changes(&self) -> BoxStream<'static, ()> {
+        Box::pin(
+            self.notifier
+                .subscribe_all_operations()
+                .await
+                .filter_map(|state| async move {
+                    match state {
+                        DummyStateMachine::OutputDone(_, _) => Some(()),
+                        DummyStateMachine::Input { .. } => Some(()),
+                        DummyStateMachine::Refund(_) => Some(()),
+                        _ => None,
+                    }
+                }),
+        )
+    }
+
+    async fn handle_cli_command(
+        &self,
+        client: &Client,
+        args: &[ffi::OsString],
+    ) -> anyhow::Result<serde_json::Value> {
+        if args.is_empty() {
+            return Err(anyhow::format_err!(
+                "Expected to be called with at least 1 arguments: <command> …"
+            ));
+        }
+
+        let command = args[0].to_string_lossy();
+
+        match command.as_ref() {
+            "print-money" => {
+                if args.len() != 2 {
+                    return Err(anyhow::format_err!(
+                        "`print-money` command expects 1 argument: <amount-msats>"
+                    ));
+                }
+
+                Ok(serde_json::to_value(
+                    client
+                        .print_money(Amount::from_str(&args[1].to_string_lossy())?)
+                        .await?,
+                )?)
+            }
+            command => Err(anyhow::format_err!(
+                "Unknown command: {command}, supported commands: print-money"
+            )),
+        }
+    }
+}
```

Comment:
Only this implementation for `handle_cli_command` is new; everything else in the `fedimint-dummy-*` crates is copied from upstream 0.1